File: POCAppJava/app/src/main/java/com/poc/pocappjava/interfaces/ListingInterface.java
package com.poc.pocappjava.interfaces;

import com.poc.pocappjava.model.Feed;

/**
 * Created by akhilmalik on 08/02/18.
 */

public interface ListingInterface {

    // function to send data fetched from server to view
    void onDataLoaded(Feed feed);

    // call back for reporting error
    void onError(String error);
}

File: app/src/main/java/com/hcl/poc/interfaces/ListingInterface.kt
package com.hcl.poc.interfaces

import com.hcl.poc.model.Feed

/**
 * Created by akhilmalik on 07/02/18.
 *
 * Interface for getting callbacks of webservice response
 */
interface ListingInterface {
    // function to send data fetched from server to view
    fun onDataLoaded(feed: Feed)

    // call back for reporting error
    fun onError(error: String)
}